dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/udp/UDPUtils.java

```diff
@@ -0,0 +1,256 @@
+                }
+            } else {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "bindHelper failed due to: " + future.cause().getMessage());
+                }
+
+                if (retryCount > 0) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                        Tr.debug(tc, "attempt to bind again after a wait of " + retryDelay + "ms; " + retryCount
+                                + " attempts remaining");
+                    }
+                    // recurse until we either complete successfully or run out of retries;
+                    try {
+                        Thread.sleep(retryDelay);
+                    } catch (InterruptedException x) {
+                        // do nothing but debug
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc, "sleep caught InterruptedException.  will proceed.");
+                        }
+                    }
+                    bind(framework, bootstrap, newHost, inetPort, retryCount - 1, retryDelay, bindListener);
+                } else {
+                    if (config.isInboundChannel()) {
+                        Tr.error(tc, UDPMessageConstants.BIND_FAILURE,
+                                new Object[] { channelName, newHost, String.valueOf(inetPort) });
+                    } else {
+                        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                            Tr.debug(tc, UDPMessageConstants.BIND_FAILURE,
+                                    new Object[] { channelName, newHost, String.valueOf(inetPort) });
+                        }
+                    }
+                }
+            }
+        });
+        return bindFuture;
+    }
+
+    /**
+     * Start an outbound UDP channel
+     * 
+     * @param nettyFrameworkImpl
+     * @param bootstrap
+     * @param inetHost
+     * @param inetPort
+     * @param bindListener
+     * @return
+     * @throws NettyException
+     */
+    public static ChannelFuture startOutbound(NettyFrameworkImpl framework, BootstrapExtended bootstrap,
+            String inetHost, int inetPort, ChannelFutureListener bindListener) throws NettyException {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "startOutbound (UDP): attempt to bind a channel at host " + inetHost + " port " + inetPort);
+        }
+        return startHelper(framework, bootstrap, inetHost, inetPort, bindListener);
+    }
+
+    /**
+     * Start an inbound UDP channel
+     * 
+     * @param framework
+     * @param bootstrap
+     * @param inetHost
+     * @param inetPort
+     * @param bindListener
+     * @return
+     * @throws NettyException
+     */
+    public static ChannelFuture start(NettyFrameworkImpl framework, BootstrapExtended bootstrap, String inetHost,
```

Comment:
Would it be better startInbound?

Reply:
There's a couple of references that can definitely be clarified/updated. I would like to do this work while working on updating the SIP Netty changes so created https://github.com/OpenLiberty/open-liberty/issues/24840 to follow up on this